catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java

```diff
@@ -212,6 +213,68 @@
 
   private String customBackgroundSlideout;
 
+  private List<String> editorAttributes = Collections.emptyList();
+  private List<String> requiredAttributes = Collections.emptyList();
+  private Map<String, Set<String>> attributeEnumMap = Collections.emptyMap();
+
+  public List<String> getEditorAttributes() {
+    return editorAttributes;
+  }
+
+  public void setEditorAttributes(List<String> editorAttributes) {
+    this.editorAttributes = editorAttributes;
+  }
+
+  public List<String> getRequiredAttributes() {
+    return requiredAttributes;
+  }
+
+  public void setRequiredAttributes(List<String> requiredAttributes) {
+    this.requiredAttributes = requiredAttributes;
+  }
+
+  public Map<String, Set<String>> getAttributeEnumMap() {
+    return attributeEnumMap;
+  }
+
+  public void setAttributeEnumMap(Map<String, Set<String>> attributeEnumMap) {
+    this.attributeEnumMap = attributeEnumMap;
+  }
+
+  public void setAttributeEnumMap(List<String> entries) {
+    List<String> displayedAttributes = new ArrayList<>();
+    Map<String, Set<String>> attributeValueEnums = new HashMap<>();
+
+    for (String entry : entries) {
+      if (entry.isEmpty()) {
+        continue;
+      }
+
+      String[] kvPair = entry.split("=", 2);
+      String attrName = kvPair[0];
```

Comment:
❗️ I think you could still get exceptions if you don't check the values or length of `kvPair` before accessing them.

Reply:
I didn't consider lists with null strings because that never happens with metatype configuration. I'll add a check though to be safe. 

With nonnull strings, though, the split method will always produce an array with at least one element. And I only access the second after checking the length, so this should be safe.